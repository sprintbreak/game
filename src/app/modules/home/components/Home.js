import React from 'react'
import './Home.scss'
import Button from '@santander/everest-ui/lib/Button'
import propTypes from 'prop-types'

export const Home = (props) => (
    <>
        <div className='customClass'>This is a dumb component</div>
        <div>
            The button below has the SantanderText-Regular font-family. You DONT
            need to put any fonts in the project. See index.js imports and find
            fonts.scss from everest. You can use all fonts provided by ux for
            free. Also, See the icons.scss import, to use all icons in our set.
            See more :
            https://everest-storybook-everest-ui-storybook.apps.ocppaz1.ar.bsch/?path=/story/components-icon--icon-set
        </div>
        <Button
            text='Hello!'
            options='centered'
            size='small'
            type='primary'
            disabled={false}
            onClick={() => props.history.push('/withState')}
            icon={{
                icon: 'icon-SYS001',
                position: 'right'
            }}
        />
    </>
)

Home.propTypes = {
    history: propTypes.object
}

export default Home
