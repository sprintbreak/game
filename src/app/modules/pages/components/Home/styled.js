import styled from 'styled-components';

export const PageContainer = styled.div`
    height: 100%;
    box-sizing: border-box;
`;

export const PageHome = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    height: 100%;
`;

export const ImgHome = styled.div`
    margin: 5rem;

    img {
        border-bottom: 1px solid #e72727;
        /* box-shadow: 0 4px 2px -2px rgba(0, 0, 0, 0.1); */
    }
`;

export const BtnHome = styled.div`

    /* .main_btn {
        position: relative;
        display: inline-block;
        background: #e72727;
        padding: 0px 40px;
        color: #fff;
        font-size: 13px;
        font-weight: 700;
        line-height: 60px;
        border: 0px;
        outline: none !important;
        box-shadow: none !important;
        text-align: center;
        cursor: pointer;
        border-radius: 0px;
        text-transform: uppercase;
        letter-spacing: 0.5px;
        z-index: 1;
        transition: all 300ms linear 0s;
        @media (max-width: 767px) {
            line-height: 50px;
        }
        &:before {
            content: '';
            position: absolute;
            top: 5px;
            left: 5px;
            bottom: 5px;
            right: 5px;
            background: #e72727;
            border: 1px solid #ffffff;
            z-index: -1;
            transition: all 300ms linear 0s;
        }
        &:after {
            content: '';
            position: absolute;
            bottom: 5px;
            right: 5px;
            border-right: 10px solid #ffffff;
            border-left: 10px solid transparent;
            border-bottom: 10px solid #ffffff;
            border-top: 10px solid transparent;
            transition: all 300ms linear 0s;
        }
        &:hover {
            color: #ffffff;
            background: #262533;
            &:before {
                background: #262533;
            }
            &:after {
                border-right: 10px solid #ffffff;
                border-bottom: 10px solid #ffffff;
            }
        }
    } */
`;