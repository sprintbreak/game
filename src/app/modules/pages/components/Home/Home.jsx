import React, { useEffect, useState } from 'react'
import config from '../../../../config/config';
import { BtnHome, PageHome, ImgHome, PageContainer } from './styled';
import Button from './../../../components/Button/Button';
import { connect } from 'react-redux';
import Header from './../../../components/Header/Header';
import imgHome from './../../../../../assets/img/juego_crop.png';
import { authenticateWs, logout } from './../../../store/actions/loginActions';
import { joinRoom, leaveRoom } from './../../../store/actions/roomActions';
import { initializePlayer } from './../../../store/actions/playerActions';
import { wsDispatch } from './../../../store/actions/wsActions';
import { useHistory } from 'react-router-dom';
import Websocket from 'react-websocket';
import { Alert } from '@material-ui/lab'

const Home = props => {

    const { id, error, dispatchWs, session, initializePlayer, joinRoom, inRoom } = props;

    const history = useHistory();
    const ws = React.createRef();

    useEffect(() => {
        initializePlayer(id);
    }, [])

    // useEffect(() => {
    //     console.log(error);
    // }, [error])

    useEffect(() => {
        if(inRoom) history.replace("/room", ws.current);
    }, [inRoom])

    const handleJoinRoom = () => {
        joinRoom(id, session, ws.current);
    }

    const wsOpen = () => authenticateWs(ws.current, session);

    const wsClose = data => console.log('WS Close', data);

    const wsData = data => {
        console.log('WS Recieved: ', data);
        dispatchWs(id, data, { props, ws: ws.current });
    }

    const wsDataError = (data) => {
        console.error("WS Error", data)
    }

    return (
        <PageContainer>
            <Websocket url={config.api.ws_url}
                debug={false}
                reconnect={true}
                onOpen={wsOpen}
                onClose={wsClose}
                onMessage={wsData}
                onError={wsDataError}
                ref={ws}
            />
            <Header />
            <PageHome>
                <ImgHome>
                    <img src={imgHome} alt="cartas-home" width="600" />
                </ImgHome>
                <BtnHome>
                    <Button className="main_btn" variant="outlined" onClick={handleJoinRoom}>Entrar al juego</Button>
                </BtnHome>
                { error && <Alert severity="error">{error}</Alert> }
            </PageHome>
        </PageContainer>
    )
}

const mapStateToProps = state => {
    return {
        error: state.appReducer.error,
        id: state.appReducer.id,
        inRoom: state.appReducer.inRoom,
        session: state.appReducer.session,
    }
}

const mapDispatchToProps = dispatch => {
    return {
        authenticateWs: (ws, session) => dispatch(authenticateWs(ws, session)),
        dispatchWs: (id, data, { props, ws }) => dispatch(wsDispatch(id, data, { props, ws })),
        initializePlayer: (id) => dispatch(initializePlayer(id)),
        joinRoom: (id, session, ws) => dispatch(joinRoom(id, session, ws)),
        leaveRoom: (id, session, ws) => dispatch(leaveRoom(id, session, ws)),
        logout: (id, session) => dispatch(logout(id, session)),
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Home);
