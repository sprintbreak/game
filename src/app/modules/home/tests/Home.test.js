import React from 'react'
import Home from '../components/Home'
import renderer from 'react-test-renderer'
import { theme } from '@santander/everest-ui'
import { ThemeProvider } from 'styled-components'

it('renders correctly', () => {
    const tree = renderer
        .create(
            <ThemeProvider theme={theme}>
                <Home />
            </ThemeProvider>
        )
        .toJSON()
    expect(tree).toMatchSnapshot()
})
