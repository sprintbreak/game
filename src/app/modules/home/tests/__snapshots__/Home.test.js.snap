// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="customClass"
  >
    This is a dumb component
  </div>,
  <div>
    The button below has the SantanderText-Regular font-family. You DONT need to put any fonts in the project. See index.js imports and find fonts.scss from everest. You can use all fonts provided by ux for free. Also, See the icons.scss import, to use all icons in our set. See more : https://everest-storybook-everest-ui-storybook.apps.ocppaz1.ar.bsch/?path=/story/components-icon--icon-set
  </div>,
  <button
    className="styled__StyledButton-kdcIry fqYQOA"
    disabled={false}
    onClick={[Function]}
    size="small"
    type="primary"
  >
    Hello!
    <span
      className="icon-SYS001"
      style={
        Object {
          "color": "inherit",
          "fontSize": "24px",
        }
      }
    />
  </button>,
]
`;
